code/datums/weakrefs.dm

```diff
@@ -53,18 +53,15 @@
  * will always hard delete.
  */
 /datum/weakref
-	var/reference
+	VAR_PRIVATE/datum/thing
 
 /datum/weakref/New(datum/thing)
-	reference = REF(thing)
+	src.thing = thing
+	RegisterSignal(thing, COMSIG_PARENT_QDELETING, PROC_REF(on_qdeleting))
 
 /datum/weakref/Destroy(force)
 	var/datum/target = resolve()
 	qdel(target)
-
-	if(!force)
-		return QDEL_HINT_LETMELIVE //Let BYOND autoGC thiswhen nothing is using it anymore.
-	target?.weak_reference = null
```

Comment:
This code was never necessary because `/datum/Destroy` clears the weakref.

Reply:
this code is nessesary and you are wrong

Reply:
weakrefs are weak because they can exist past their target being deleted. you can't "own" a weakref, which is why we require force to del one.

Reply:
I willf ix it